src/repr/src/adt/mz_acl_item.rs

```diff
@@ -371,6 +376,146 @@ impl Columnation for MzAclItem {
     type InnerRegion = CloneRegion<MzAclItem>;
 }
 
+/// A list of privileges granted to a role.
+///
+/// This is primarily used for compatibility in PostgreSQL.
+///
+/// See: <https://github.com/postgres/postgres/blob/7f5b19817eaf38e70ad1153db4e644ee9456853e/src/include/utils/acl.h#L48-L59>
+#[derive(
+    Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Serialize, Hash, Deserialize, Arbitrary,
+)]
+pub struct AclItem {
+    /// Role that this item grants privileges to.
+    pub grantee: Oid,
+    /// Grantor of privileges.
+    pub grantor: Oid,
+    /// Privileges bit flag.
+    pub acl_mode: AclMode,
+}
+
+impl AclItem {
+    pub fn empty(grantee: Oid, grantor: Oid) -> AclItem {
+        AclItem {
+            grantee,
+            grantor,
+            acl_mode: AclMode::empty(),
+        }
+    }
+}
+
+// PostgreSQL has no stable binary encoding for AclItems. However, we need to be able to convert
+// types to and from binary, so we invent our own encoding. The encoding matches the in memory
+// format that PostgreSQL uses, which is (Oid, Oid, AclMode).
+impl AclItem {
+    pub fn encode_binary(&self) -> Vec<u8> {
```

Comment:
I don't know much about our pgwire encoding, but this makes me wonder if it would be worth while to have a `trait BinaryEncode` that we implement for our types. For example, the detail that `Oid`s are encoded as little endian bytes probably shouldn't live here.